.github/workflows/spelling.yml

```diff
@@ -0,0 +1,31 @@
+name: Spell checking
+on:
+  pull_request_target:
+  push:
+  issue_comment:
+    types: [created]
+
+jobs:
+  spelling:
+    name: Spell checking
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout-merge
+      if: "contains(github.event_name, 'pull_request')"
+      uses: actions/checkout@v2
+      with:
+        ref: refs/pull/${{github.event.pull_request.number}}/merge
+    - name: checkout
+      if: ${{ github.event_name == 'push' ||
+        (
+          contains(github.event.comment.body, '@check-spelling-bot apply')
+        ) }}
+      uses: actions/checkout@v2
+    - uses: check-spelling/check-spelling@main
```

Comment:
This should be tied to a specific version rather than `main` -- probably `0.0.19` in this case. Quoting the quick start guide you linked:

> Best-practice for workflows is to use a pinned version.